File: 2028FindMissingObservations.py
class Solution(object):
    def missingRolls(self, rolls, mean, n):
        """
        :type rolls: List[int]
        :type mean: int
        :type n: int
        :rtype: List[int]
        """
        ans=[]
        sum_roll=len(rolls)+n
        lack=(mean*sum_roll)-sum(rolls)
        if lack > n*6 or lack<0:
            return []
        else:
            ans=[lack/n for i in range(0,n)]
            if lack%n!=0:
                print lack%n
                for i in range(0, lack%n):
                    ans[i]+=1
            if 0 in ans:
                return []
            return ans

File: 2028FindMissingObservations.cpp
class Solution {
public:
    vector<int> missingRolls(vector<int>& rolls, int mean, int n) {
        int sum_roll = rolls.size() + n;
        int sum=0;
        for (int i = 0; i<rolls.size();i++){
            sum+=rolls[i];
        }
        int lack = (sum_roll*mean)-sum;
        vector<int> ans;
        if (lack>n*6 || lack <0||lack/n==0){
            return ans;
        }
        else{
            for (int i=0;i<n;i++){
                ans.push_back(lack/n);
            }
            if (lack%n!=0){
                for (int i=0;i<lack%n;i++){
                    ans[i]+=1;
                }
            }
        }
        return ans;
    }
};
